frontend/src/ClientDataForm.js

```diff
@@ -117,6 +123,36 @@ const ClientDataForm = () => {
     </div>
   );
 
+  if (loading) {
+    return (
+      <>
+        <div className="container d-flex justify-content-center align-items-center min-vh-100">
```

Comment:
We should look into making this spinner it's own component so it can be used on other pages while loading.
I added an issue on our to-do.